api/v1alpha3/gcpmachine_types.go

```diff
@@ -64,6 +64,11 @@ type GCPMachineSpec struct {
 	// +optional
 	AdditionalLabels Labels `json:"additionalLabels,omitempty"`
 
+	// AdditionalMetadata is an optional set of metadata to add to an instance, in addition to the ones added by default by the
+	// GCP provider.
+	// +optional
+	AdditionalMetadata []MetadataItem `json:"additionalMetadata,omitempty"`
```

Comment:
I think this should probably also have some markers set to indicate that additionalMetadata should be treated as type `x-kubernetes-list-type` of `map`:
```suggestion
	// AdditionalMetadata is an optional set of metadata to add to an instance, in addition to the ones added by default by the
	// GCP provider.
	// +listType=map
	// +listMapKey=key
	// +optional
	AdditionalMetadata []MetadataItem `json:"additionalMetadata,omitempty"`
```

Reply:
@justinsb can you take a look at Jason's comment when you have a chance?

Reply:
Done - great point @detiber  - and thanks for the ping @cpanato !